src/request/swap-iframe/SwapIFrameApi.js

```diff
@@ -0,0 +1,385 @@
+                            input => ({
+                                ...input,
+                                witnessUtxo: {
+                                    ...input.witnessUtxo,
+                                    script: BitcoinJS.Buffer.from(input.witnessUtxo.script, 'hex'),
+                                },
+                            }),
+                        ),
+                    },
+                } : {}
+            ),
+        };
+
+        // Parse request
+        /** @type {Parsed<KeyguardRequest.SignSwapTransactionsRequest>} */
+        // @ts-ignore Missing the following properties: fund, redeem
+        const parsedRequest = {
+            swapId: request.swapId,
+        };
+
+        if (storedRequest.fund.type !== request.fund.type || storedRequest.redeem.type !== request.redeem.type) {
+            throw new Errors.InvalidRequestError('Different swap assets in iframe request than in top-level request');
+        }
+
+        if (request.fund.type === 'NIM' && storedRequest.fund.type === 'NIM') {
+            const htlcDetails = HtlcUtils.decodeNimHtlcData(request.fund.htlcData);
+
+            if (htlcDetails.refundAddress !== storedRequest.fund.transaction.sender.toUserFriendlyAddress()) {
+                throw new Errors.InvalidRequestError('NIM HTLC refund address must be same as sender');
+            }
+
+            // Check that validityStartHeight is before HTLC timeout
+            if (storedRequest.fund.transaction.validityStartHeight >= htlcDetails.timeoutBlockHeight) {
+                throw new Errors.InvalidRequestError(
+                    'Fund validityStartHeight must be lower than HTLC timeout block height',
+                );
+            }
+
+            parsedRequest.fund = {
+                type: 'NIM',
+                htlcDetails,
+                htlcData: request.fund.htlcData,
+            };
+        }
+
+        if (request.redeem.type === 'NIM' && storedRequest.redeem.type === 'NIM') {
+            const htlcDetails = HtlcUtils.decodeNimHtlcData(request.redeem.htlcData);
+
+            if (htlcDetails.redeemAddress !== storedRequest.redeem.transaction.recipient.toUserFriendlyAddress()) {
+                throw new Errors.InvalidRequestError('NIM HTLC redeem address must be same as recipient');
+            }
+
+            // Check that validityStartHeight is before HTLC timeout
+            if (storedRequest.redeem.transaction.validityStartHeight >= htlcDetails.timeoutBlockHeight) {
+                throw new Errors.InvalidRequestError(
+                    'Redeem validityStartHeight must be lower than HTLC timeout block height',
+                );
+            }
+
+            parsedRequest.redeem = {
+                type: 'NIM',
+                htlcDetails,
+                htlcData: request.redeem.htlcData,
+                htlcAddress: this.parseAddress(request.redeem.htlcAddress, 'redeem.htlcAddress')
+                    .toUserFriendlyAddress(),
```

Comment:
I think that is unnecessary, because if the HTLC address actually isn't this address, the swap does not get passed step 2 in the wallet.

And as a precaution against the HTLC address getting faked in the request: a whole transaction can just as easily be faked - the Keyguard cannot validate if a transaction is included in the chain or not.